src/nfa/operations.cc

```diff
@@ -15,6 +15,9 @@
  * GNU General Public License for more details.
  */
 
+//is it good that header file names don't match cc files? What is the system actually? Is there a difference between operations and algorithms?
```

Comment:
The idea is that `algorithms.hh` is a header file for a a few specific algorithms over NFA, such as minimization, complementation, etc. with all of their different variations using different implementations. That is why we have  a dispatch function for each such algorithm in a `nfa.hh` and if you do not want to use the dispatch function, you need to include `algorithms.hh` file.

`operations.cc` is a more general source file for definition of various functions, some of which might not be in `algorithms.hh`. Ideally, `operations.cc` would contain definitions of all functions in `nfa.hh` (and now in `algorithms.hh`) and `nfa.cc` would contain only definitions for methods of `Nfa`.

We can consider creating a source file for `algorithms.hh`, called `algorithms.cc` and keep in `operations.cc` only definitions of function over NFAs from `nfa.hh`. This should split the source files into reasonably large files. 

What are some other ideas?

Reply:
ok, we already talked this through